From 7c4d019a306d2dd808a0b395d9ef3f94701e57ac Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 15 Feb 2022 12:32:14 +0000
Subject: [PATCH 1057/1076] MANTIS 0061609 change: Assure ID - Detect and
 report document class via IdDocument struct

OneIdTypes.xsd
new field documentClass


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1701 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Cuss/OneIdTypes.xsd            | 51 ++++++++++++++++++-
 1 file changed, 50 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd b/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
index 37460074..3d837531 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
+++ b/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
@@ -51,6 +51,35 @@
 			<xs:enumeration value="enrolFailed"/>
 			<xs:enumeration value="acceptanceDenied"/>
 			<xs:enumeration value="platformError"/>
+			<xs:enumeration value="invalidIdDocument"/>
+			<xs:enumeration value="invalidBoardingPass"/>
+			<xs:enumeration value="skipped"/>
+			<xs:enumeration value="backendError"/>
+		</xs:restriction>
+	</xs:simpleType>
+	
+	<xs:simpleType name="documentClass">
+		<xs:annotation>
+			<xs:documentation>The document class of the scanned document</xs:documentation>
+		</xs:annotation>
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="BirthCertificate"/>
+			<xs:enumeration value="ConsularIdentification"/>
+			<xs:enumeration value="Currency"/>
+			<xs:enumeration value="DriversLicense"/>
+			<xs:enumeration value="IdentificationCard"/>
+			<xs:enumeration value="Military"/>
+			<xs:enumeration value="Other"/>
+			<xs:enumeration value="Passport"/>
+			<xs:enumeration value="Permit"/>
+			<xs:enumeration value="ResidenceDocument"/>
+			<xs:enumeration value="TravelDocument"/>
+			<xs:enumeration value="TribalIdentification"/>
+			<xs:enumeration value="Unknown"/>
+			<xs:enumeration value="VehicleRegistration"/>
+			<xs:enumeration value="Visa"/>
+			<xs:enumeration value="VoterIdentification"/>
+			<xs:enumeration value="WeaponLicense"/>			
 		</xs:restriction>
 	</xs:simpleType>
 	
@@ -265,6 +294,11 @@
 					<xs:documentation>The textual description of the id-document if available, e.g. 'Germany (D) Identity Card'. This data is for instance returned by the AssureID-SDK while parsing ID documents (IDDR) via a Desko scanner</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="documentClass" type="oneIdTypes:documentClass" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>The document class of the id-document if available, e.g. 'Passport'. This data is for instance returned by the AssureID-SDK while parsing ID documents (IDDR) via a Desko scanners</xs:documentation>
+				</xs:annotation>
+			</xs:element>
 		</xs:sequence>
 	</xs:complexType>
 	<xs:complexType name="boardingPassData">
@@ -368,7 +402,12 @@
 			</xs:element>
 			<xs:element name="idDocument" type="oneIdTypes:idDocumentData" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>TBC</xs:documentation>
+					<xs:documentation>ID document data read from the 1:n identificaiton or by a physical passport scan (this semantic was changed from interface version 1.1 to 1.2!)</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="idDocumentBooking" type="oneIdTypes:idDocumentData" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>ID document data read from the booking / DCS / PNR</xs:documentation>
 				</xs:annotation>
 			</xs:element>
 			<xs:element name="boardingPass" type="oneIdTypes:boardingPassData" minOccurs="0">
@@ -376,6 +415,16 @@
 					<xs:documentation>TBC</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="fqtvNumber" type="xs:string" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>The passenger's frequent fligher number</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="selectee" type="xs:boolean" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Indicates if a passenger is a 'selectee' and may require special handling in bio sub-process</xs:documentation>
+				</xs:annotation>
+			</xs:element>
 			<xs:element name="facialImage" minOccurs="0">
 				<xs:annotation>
 					<xs:documentation>Base64 encoded facial image data. Required in case of identification via facial image</xs:documentation>
-- 
2.41.0.windows.1

